### PR TITLE
polish(theme): better error messages on navbar item rendering failures + ErrorCauseBoundary API

### PR DESCRIPTION
**Pull request by tannerdolby**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Improves the DocNavbarItem error messages to help new users understand that the Docusaurus config allows them to avoid the issue mentioned in https://github.com/facebook/docusaurus/discussions/8692 by updating `themeConfig.navbar.items`.

### Test links

Deploy preview: https://deploy-preview-8735--docusaurus-2.netlify.app/

## Related issues/PRs

https://github.com/facebook/docusaurus/issues/8696

### CODE REVIEW

**Review by slorber — CHANGES_REQUESTED** [review]

File: packages/docusaurus-theme-common/src/utils/docsUtils.tsx [hunk 1]
```diff
@@ -345,7 +345,9 @@ export function useLayoutDoc(
       throw new Error(
         `DocNavbarItem: couldn't find any doc with id "${docId}" in version${
           versions.length > 1 ? 's' : ''
-        } ${versions.map((version) => version.name).join(', ')}".
+        } "${versions.map((version) => version.name).join(', ')}".
+Check themeConfig.navbar.items in your Docusaurus config to make sure
+navbar items aren't attempting to reference a non-existent doc.
```
[the remark this answers is not in the document]
Comment (tannerdolby) [reply packages/docusaurus-theme-common/src/utils/docsUtils.tsx]: Gotcha! That makes sense. I've updated things to reflect your suggestions from https://github.com/facebook/docusaurus/issues/8696#issuecomment-1443395348

**Comment by slorber** [comment]
Thanks

Did some modifications because the error page wasn't able to render the full error causal chain.

Now it will print everything:

CleanShot 2023-03-09 at 17 55 31

Not super pretty nor great DX, but message is clearer than before.

**Comment by slorber** [comment]
Also found a generic solution to apply this automatically to all navbar items. It should make it easier for users to debug issues.

CleanShot 2023-03-09 at 18 18 02

**Comment by tannerdolby** [comment]
> Also found a generic solution to apply this automatically to all navbar items. It should make it easier for users to debug issues.

Nice work! Your generic solution works better and provides users a much easier way to debug the errors.

**Pull request merged**